### DISCUSSION TITLE
PathLayer style doesn't change

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1123] [Q&A] [answered]
I create a `PathLayer` with his `Style` and add it on `map`:

```
Style style = Style.builder()
            .generalization(Style.GENERALIZATION_MEDIUM)
            .strokeColor(Color.BLUE)
            .strokeWidth(5f)
            .cap(Paint.Cap.ROUND)
            .build();

pathLayer = new PathLayer(map, style);

map.layers().add(pathLayer);
map.clearMap();
```
so far everything is ok.

So, I want to change the `Color` of this `PathLayer` later, so:
```
Style style = Style.builder()
            .generalization(Style.GENERALIZATION_MEDIUM)
            .strokeColor(Color.GREEN)
            .strokeWidth(5f)
            .cap(Paint.Cap.ROUND)
            .build();
                       
pathLayer.setStyle(style);
```

but nothing happens.

I have tired `map.update(true)`, `map.render()`, `map.clearMap()`

**Comment by devemux86** [comment]
It is a new layer with a new appearance. Remove the old layer and add the new one.
Can also try the setPoints.

**Reply by moiatto** [reply]
Ok. Thanks
I'll try.

**Reply by moiatto** [reply, marked as answer]
For those who have the same problem as me, this is the solution:
```
PathLayer newPathLayer = new PathLayer(map, newStyle));
newPathLayer.setPoints(pathLayer.getPoints());
map.layers().set(map.layers().indexOf(pathLayer), newPathLayer);
```

**Comment by moiatto** [comment]
@devemux86 
sorry, a different question than this:
the names of streets, cities, etc. with their coordinates are saved in the map file (.map) or in the POI file (.poi) ?
Could you suggest me an example file to follow?
I would like to perform an offline search.

Many thanks

**Reply by devemux86** [reply]
For offline search you can use the poi files. See the PoiSearchActivity example in the samples.

**Reply by moiatto** [reply]
Thanks again
